lib/store_web/live/cart_live/index.ex

```diff
@@ -54,14 +55,42 @@ defmodule StoreWeb.CartLive.Index do
       |> Repo.one()
       |> Repo.preload(:products)
 
+    quantity = Enum.count(order.products)
+
+    if quantity == 1 do
+      order
+      |> Order.changeset(%{status: :canceled})
+      |> Repo.update!()
+    else
+      order_product =
+        OrdersProducts
+        |> where(order_id: ^order.id)
+        |> where(product_id: ^id)
+        |> Repo.one()
```

Comment:
These functions that use database queries should be abstracted to the context module. This way you are _disrupting_ the model-view-controller architecture.